test/unit/atomic/test-atomic-ref-accessors.cpp

```diff
@@ -0,0 +1,139 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-19, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+///
+/// Source file containing tests for atomic accessor methods
+///
+
+#include <RAJA/RAJA.hpp>
+#include "RAJA_gtest.hpp"
+
+#if defined(RAJA_ENABLE_CUDA)
+#include "RAJA_unit_forone.hpp"
+#endif
+
+template <typename T, typename AtomicPolicy>
+void testAtomicAccessors()
+{
+  // should also work with CUDA
+  T theval = (T)0;
+  T * memaddr = &theval;
+
+  // explicit constructor with memory address
+  RAJA::AtomicRef<T, AtomicPolicy> test1( memaddr );
+
+  // test store method with op()
+  test1.store( (T)19 );
+  ASSERT_EQ( test1, 19 );
+
+  // test assignment operator
+  test1 = (T)23;
+  ASSERT_EQ( test1, 23 );
+
+  // test load method
+  test1 = (T)29;
+  ASSERT_EQ( test1.load(), 29 );
+}
+
+// Pure CUDA test.
+#if defined(RAJA_ENABLE_CUDA)
+template <typename T, typename AtomicPolicy>
+void testAtomicAccessorsCUDA()
+{
+  T * memaddr = nullptr;
+  T * result = nullptr;
+  cudaErrchk(cudaMallocManaged((void **)&memaddr, sizeof(T)));
+  cudaErrchk(cudaMallocManaged((void **)&result, sizeof(T)));
+  cudaErrchk(cudaDeviceSynchronize());
+
+  // explicit constructor with memory address
+  RAJA::AtomicRef<T, AtomicPolicy> test1( memaddr );
+
+  // test store method with op()
+  forone<<<1,1>>>( [=] __device__ () {test1.store( (T)19 );} );
+  cudaErrchk(cudaDeviceSynchronize());
+  ASSERT_EQ( test1, (T)19 );
+
+  // test assignment operator
+  forone<<<1,1>>>( [=] __device__ () {test1 = (T)23;} );
+  cudaErrchk(cudaDeviceSynchronize());
+  ASSERT_EQ( test1, (T)23 );
+
+  // test load method
+  forone<<<1,1>>>( [=] __device__ () {test1 = (T)29; result[0] = test1.load();} );
```

Comment:
should this be `result[0] = (test1 = (T)29);`

Reply:
I wanted to test the load() method in particular, but I suppose adding an op() test like yours will help too.

Reply:
this should have an operator T test as well, a.k.a. `result[0] = test1;`